### PR TITLE
vz-11713, Check Availability of components

### PR DESCRIPTION
**Pull request by AbeHern**
Additional logic to check for availability of components

### CODE REVIEW

File: tools/vz/cmd/analyze/analyze_test.go [hunk 1]
```diff
@@ -265,3 +267,27 @@ func TestAnalyzeCommandVZTarGZFile(t *testing.T) {
...
+	assert.NotNil(t, cmd)
+	cmd.PersistentFlags().Set(constants.DirectoryFlagName, mysqlUnavailable)
+	cmd.PersistentFlags().Set(constants.ReportFormatFlagName, constants.SummaryReport)
+	err := cmd.Execute()
+	assert.Nil(t, err)
+	buf, err := os.ReadFile(rc.Out.Name())
+	assert.NoError(t, err)
+	assert.Contains(t, string(buf), unavailableErr)
+
+	// Failures must be reported underreport file details-XXXXXX.out
```
Comment (marcogelle) [inline tools/vz/cmd/analyze/analyze_test.go:288]: typo: "underreport" -> "under report"

File: tools/vz/pkg/internal/util/cluster/install.go [hunk 2]
```diff
@@ -365,34 +371,35 @@ func analyzeIstioIngressService(log *zap.SugaredLogger, clusterRoot string, issu
 }
 
 // Read the Verrazzano resource and return the list of components which did not reach Ready state
-func getComponentsNotReady(log *zap.SugaredLogger, clusterRoot string) ([]string, error) {
+func getComponentsNotReady(log *zap.SugaredLogger, clusterRoot string) ([]string, []string, error) {
```
Comment (marcogelle) [inline tools/vz/pkg/internal/util/cluster/install.go:374]: It'd be nice to update the comment above this function.

File: tools/vz/pkg/internal/util/cluster/install.go [hunk 3]
```diff
@@ -404,31 +411,28 @@ func getComponentsNotReady(log *zap.SugaredLogger, clusterRoot string) ([]string
...
-		return compsNotReady, nil
 	}
-
-	return compsNotReady, nil
+	return compsNotReady, compsReadyNotAvailable, nil
 }
 
 // Read the platform operator log, report the errors found for the list of components which fail to reach Ready state
-func reportInstallIssue(log *zap.SugaredLogger, clusterRoot string, compsNotReady []string, issueReporter *report.IssueReporter) error {
+func reportInstallIssue(log *zap.SugaredLogger, clusterRoot string, compsNotReady []string, issueReporter *report.IssueReporter, notReadyComponentsFound bool) error {
```
Comment (marcogelle) [inline tools/vz/pkg/internal/util/cluster/install.go:435]: Also update the comment about this function.

Comment (AbeHern) [reply tools/vz/pkg/internal/util/cluster/install.go:435]: Updated

File: tools/vz/cmd/analyze/analyze_test.go (hunk 1 above)
Comment (AbeHern) [reply tools/vz/cmd/analyze/analyze_test.go]: Fixed

**Review by Abhijithang — APPROVED** [review]

**Review by marcogelle — APPROVED** [review]

**Pull request merged**